src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -97,14 +97,14 @@ private static async Task<Document> FixWebApplicationBuilder(Diagnostic diagnost
 
             if (originalInvocation.ArgumentList.Arguments.Count == 0)
             {
-                return editor.OriginalDocument;
+                return document;
             }
 
-            var initArgumentExpr = originalInvocation.ArgumentList.Arguments[0].Expression; // builder => { }
+            var initArgumentExpr = originalInvocation.ArgumentList.Arguments.SingleOrDefault().Expression; // builder => { }
```

Comment:
You'll probably want to check if `SingleOrDefault()` produces `null` and return the original document if it does.